OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -81,6 +82,7 @@ public Minelayer(Actor self, MinelayerInfo info)
 		{
 			Info = info;
 			this.self = self;
+			ammoPools = self.TraitsImplementing<AmmoPool>().ToArray();
```

Comment:
Shouldn't this be something like
```suggestion
			ammoPools = self.TraitsImplementing<AmmoPool>().Where(p => p.Info.Name == Info.AmmoPoolName).ToArray();
```
? And then we drop the runtime checks for the name here and in LayMines. (The name shouldn't change during runtime.)

Reply:
Yep makes sense, also pretty sure the null check on ammoPools in LayMine isn't needed since ammoPools will always be an array.